src/useGetPosts/index.js

```diff
@@ -66,24 +66,22 @@ import { apiClient } from '../utils';
  */
 function useGetPosts( args = {}, clientId, postType ) {
 	const [ options, setOptions ] = useState( [] );
-	const [ query, setQuery ] = useState( '' );
-	const [ loading, setLoading ] = useToggle();
+	const [ query, setQuery ] = useState( [] );
+	const [ loading, setLoading ] = useToggle( true );
```

Comment:
The loading state is set of false by design upon the component mounting because we need the spinner indicator toggled (shown) as soon as any of the hook’s dependencies like post-type or clientId (in the event of block duplication) changes.

https://user-images.githubusercontent.com/8451635/148210930-fa275b79-ee8c-43fb-91f7-4dfe108acd4c.mov

Reply:
I see, thank you, @mahdiyazdani . Given that `loading` is not passed from the component that uses this hook, it also won't be possible to reset it whenever other values are changed (i.e. we're unable to achieve the same behavior).

Reply:
This issue could be circumvented by using `useState` rather than `useToggle`. I have updated this PR accordingly.

Reply:
When not provided, the initial or fallback value for the [useToggle](https://github.com/sixach/wp-react-hooks/blob/main/src/useToggle/index.js#L30) hook is considered to be primitive `false.` The loading indicator status should be toggled to ON when entered into the hook and switched back to OFF when the last `then` method gets executed. [Here](https://github.com/sixach/wp-block-twitter-feed/blob/e5bbfb0ea371857dfc58d78ab800baceddd6b5a1/src/hooks/useTweets.js#L39) is the latest usage of this hook in the Twitter block.

Reply:
One disadvantage of the usage of `useToggle` in these instances is that the component re-renders immediately after being mounted. I would thus find `setState` the more appropriate function.